src/cudamatrix/cu-matrix-inl.h

```diff
@@ -29,14 +29,20 @@ inline CuSubMatrix<Real>::CuSubMatrix(const CuMatrixBase<Real> &mat,
                                       const MatrixIndexT row_offset,
                                       const MatrixIndexT num_rows,
                                       const MatrixIndexT col_offset,
-                                      const MatrixIndexT num_cols):
-    CuMatrixBase<Real>(mat.data_ + (row_offset * mat.stride_) + col_offset,
-                       num_rows,
-                       num_cols,
-                       mat.stride_) {
-  KALDI_ASSERT(row_offset >= 0 && col_offset >= 0 &&
-               row_offset + num_rows <= mat.num_rows_ &&
-               col_offset + num_cols <= mat.num_cols_);
+                                      const MatrixIndexT num_cols) {
+  if (num_rows == 0 || num_cols == 0) {
+    KALDI_ASSERT(num_rows == 0 && num_cols == 0);
+    // Everything will have been set to zero in CuMastrixBase's default
+    // initializer, so nothing to do.
+  } else {
+    KALDI_ASSERT(row_offset >= 0 && col_offset >= 0 &&
+                 row_offset + num_rows <= mat.num_rows_ &&
+                 col_offset + num_cols <= mat.num_cols_);
+    this->data_ = mat.data_ + (row_offset * mat.stride_) + col_offset;
```

Comment:
Not sure if using `this->` with member variables improves readability, as the member variable naming pattern with the underscore `xxxx_` implies `this->`. Maybe this is just me, but when I am reading the code, I am stumbling and wondering why did they have to write `this->` explicitly. Up to you.